src/services/neo4jService.js

```diff
@@ -1,185 +1,172 @@
-import neo4j from "neo4j-driver";
+import { driver as createDriver } from "neo4j-driver";
 
-/**
- * Hooks into the neo4jDesktopApi
- *
- * Note: this integration is going to be deprecated in desktop api 2.0
- */
-const neo4jDesktopApi = window.neo4jDesktopApi;
+class Neo4JService {
+  _getNeo4jDriver = async () => {
+    if (!this.driver) {
+      try {
+        /**
+         * Hooks into the neo4jDesktopApi.
+         * Note: this integration is going to be deprecated in desktop api 2.0
+         */
+        const context = await window.neo4jDesktopApi.getContext();
 
-/**
- * Returns a neo4j driver instance from the boltUrl, username, and passwords.
- *
- * Currently accepts bolt and bolt+routing
- * @param {String} boltUrl
- * @param {String} username
- * @param {String} password
- */
-const createDriver = (boltUrl, username, password) =>
-  neo4j.driver(boltUrl, neo4j.auth.basic(username, password));
-
-export default {
-  getNeo4jDriver: async function () {
-    let driver = undefined;
-    if (neo4jDesktopApi) {
-      await neo4jDesktopApi.getContext().then((context) => {
         for (let project of context.projects) {
           for (let graph of project.graphs) {
             if (graph.status === "ACTIVE") {
-              console.log(
-                "Active graph is; " +
-                  graph.name +
-                  " (" +
-                  graph.description +
-                  ")"
-              );
+              console.log(`Active graph is: ${graph.name} - (${graph.description})`);
+
               let boltProtocol = graph.connection.configuration.protocols.bolt;
-              driver = createDriver(
+
+              // Already found and athenticated an active graph.
+              // No need to try and authenticate other graphs.
+              this.driver = createDriver(
```

Comment:
I think the syntax is incorrect here (?) Should be:
```
              this.driver = createDriver(
                  boltProtocol.url,
                  auth.basic(boltProtocol.username, boltProtocol.password)
              );
```

With the import:
```
import { driver as createDriver, auth } from "neo4j-driver";
```

Reply:
Aha! Good catch 👍🏾